app/models/form_item.rb

```diff
@@ -39,7 +39,7 @@ def _sorted_leaves(nodes)
         if form_item.is_a?(Questioning)
           form_item
         else
-          _sorted_leaves(children)
+          _sorted_leaves(children).flatten
```

Comment:
@hooverlunch I don't think this will prohibit multiple nesting. The previous implementation already does it in the form model: https://github.com/thecartercenter/elmo/blob/develop/app/models/form.rb#L74 and makes the entire array flat. I added it here to get a structure similar to: 
````ruby
[q1, [q2, q3], q4]
````
the q2, q3 are in the array to indicate that they belong to the group(s).

This is used later in the view:

https://github.com/thecartercenter/elmo/blob/1917_ODK_multiple_questions/app/views/forms/show.xml.erb#L106